src/components/utils/summary-api.ts

```diff
@@ -0,0 +1,40 @@
+import { consoleProxyFetchJSON } from './proxy';
+import { SummaryProps } from '../pipelines-overview/utils';
+import {
+  DataType,
+  TektonResultsOptions,
+  createTektonResultsSummaryUrl,
+} from './tekton-results';
+
+export type SummaryResponse = {
+  summary: SummaryProps[];
+};
+
+export const getResultsSummary = async (
+  namespace: string,
+  options?: TektonResultsOptions,
+  dataType?: DataType,
+  filter?: string,
```

Comment:
here filter property is defined in TektonResultsOptions and as an argument as well can we keep one filter so that it will be not confusing? May include datatype as part of options.